Added support for Amber18 for GCC compilers

batatasfritas opened the pull request.

batatasfritas committed:
Added support for Amber18 for GCC compilers

alalazo commented on scitasexternal/packages/amber/package.py:
```diff
@@ -35,33 +34,43 @@ class Amber(Package):
+    url = 'file:///fake/path/to/amber-16.tgz'
 
     version('16', '652e24512146e966a0a50335572ebd66')
+    version('18', 'ae5438d3f1e2d3379d5664c34193ae48')
```
Minor request for change. Every other package in Spack keeps the versions ordered from the most to the least recent. Can we swap the two lines above?

alalazo commented on scitasexternal/packages/amber/package.py:
```diff
@@ -35,33 +34,43 @@ class Amber(Package):
+    # boost-dependent part. However, AmberTools fails to compile if boost
+    # support is included, due to a bug. By putting packmol as a variant,
+    # AmberTools can be compiled in spite of this specific issue.
+    variant('packmol', default=False,
```
:+1: For the fix and the comment that explains it!

alalazo commented on scitasexternal/packages/amber/package.py:
```diff
@@ -79,12 +88,14 @@ def ambertools_configure_file(self):
+
+        run_env.prepend_path('PATH', join_path(self.wdir, 'bin'))
+
+        run_env.set('AMBERHOME', join_path(self.wdir))
```
`join_path` seems redundant here.

alalazo reviewed (dismissed):
Just a couple of very minor request.

batatasfritas committed:
Corrected version ordering and redundant join_path

batatasfritas committed:
Merge branch 'packages/amber' of https://github.com/epfl-scitas/spack-repo-externals into packages/amber

Conflicts:
	scitasexternal/packages/amber/package.py

rmsds reviewed (approved):
Looks good, all requested changes implemented.

rmsds commented on scitasexternal/packages/amber/package.py:
```diff
@@ -35,33 +34,43 @@ class Amber(Package):
+    variant('packmol', default=False,
+            description='Compiles the optional packmol_memgen')
+
+    depends_on('boost', when='+packmol')
```
It seems the packmol variant is only used to remove the boost dependency, but this will not work if there is a system boost installed.

Can we explicitly disable packmol if possible?

batatasfritas replied:
My latest commit removed this variant, because the bug has been corrected in the meantime.

In any case, explicit disabling of this component was not possible.

batatasfritas committed:
Updated to latest version

- Amber 18
- AmberTools 19
- Removed packmol variant because bug has been correct

rmsds commented on scitasexternal/packages/amber/package.py:
```diff
@@ -35,33 +35,38 @@ class Amber(Package):
-    url = 'file://fake/amber16.tgz'
+    url = 'file:///%s/amber-18.tar.bz2' % os.getcwd()
 
+    version('18.19', '29d6fa953ef4a6e0349f4a9a36d5767b871f6784aaf9c6746fc535202e67e200d0b354bdbd8cbeb2e0b087d48a13b00dc912a7124cfdc1fa540bf3a30568cb81')
```
This looks like a sha256sum, is it missing the `sha256=`?
Also, can you add a comment explaining what is the 18.19 version?

rmsds commented on scitasexternal/packages/amber/package.py:
```diff
@@ -35,33 +35,38 @@ class Amber(Package):
     depends_on('netcdf+mpi', when='+mpi')
     depends_on('netcdf-fortran ^netcdf+mpi', when='+mpi')
 
-    # FIXME: depends_on('X')
```
Doesn't it depend on X anymore (even when `+X`)?
